sigstore/dsse/_predicate.py

```diff
@@ -30,10 +31,15 @@
 
 from sigstore.dsse import Digest
 
-PREDICATE_TYPE_SLSA_v0_2 = "https://slsa.dev/provenance/v0.2"
-PREDICATE_TYPE_SLSA_v1_0 = "https://slsa.dev/provenance/v1"
 
-SUPPORTED_PREDICATE_TYPES = [PREDICATE_TYPE_SLSA_v0_2, PREDICATE_TYPE_SLSA_v1_0]
+class PredicateType(enum.Enum):
```

Comment:
This makes `str(PredicateType)` behave nicely, I believe 🙂 

```suggestion
class PredicateType(str, enum.Enum):
```

Reply:
doesn't that interfere with iterating over the enum values? Since [p for p in PredicateType] will behave differently if it's a subclass from `str` first rather than `PredicateType`.
I force-pushed an amend of the last commit, can you check if the way I wrote it is idiomatic?

Reply:
Fixed!